collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/FieldListUpdateTest.java

```diff
@@ -354,9 +359,13 @@ public void selectingARating_ShouldChangeRelevanceOfRelatedField() throws Except
         jumpToGroupWithText("Rating");
         onView(withText(startsWith("Source13"))).perform(click());
 
-        for (int i = 0; i < 10; i++) {
+        for (int i = 1; i <= 10; i++) {
```

Comment:
Definitely from the previous version but I'm not sure why this test loops. The goal of the test seems to be to check that setting a rating updates the relevance of another question. In my head the test could just do this once: check the "Target13" question doesn't exist, set a (any) rating, check it does, remove the rating, check "Target13". @lognaturel can you think of any reason we might be checking every value here?

Reply:
Yes, I too think you are right here @seadowg. Checking just once whether changing the rating makes changes to another question should suffice.